src/backend/metal/src/command.rs

```diff
@@ -144,18 +143,19 @@ impl QueueBlocker {
         }
     }
 
-    pub(crate) fn set_host_event(&mut self, event: &Arc<AtomicBool>) {
+    pub(crate) fn triage(&mut self) {
         // clean up the relevant blocks
-        for blocked in self.submissions.iter_mut() {
-            blocked.wait_events.retain(|ev| !Arc::ptr_eq(ev, event));
-            if blocked.affect_events.iter().any(|ev| Arc::ptr_eq(ev, event)) {
-                // Execution of this command buffer affects the event,
-                // we can't look further ahead.
-                break;
-            }
-        }
+        let done = {
+            let blocked = match self.submissions.first_mut() {
```

Comment:
If I understand correctly, why do we only check the first submission here?

For example, is it possible for `self.submissions[0]` to be blocked but `self.submissions[1]` to be unblocked – in which case we don't try to triage `self.submissions[1]` despite it being ready?

Reply:
We can't submit submissions[1] anyway for as long as submissions[0] is empty.
This method is not comprehensive, it's expected to be called "occasionally" and advance the state of things. So taking one submission seems fine.

Reply:
Oh ok, so these are ordered too. Sounds good :+1: